react/features/recording/components/Recording/StartRecordingDialogContent.js

```diff
@@ -0,0 +1,216 @@
+     *
+     * @protected
+     * @returns {React$Component}
+     */
+    render() {
+        const { isTokenValid, isValidating, t } = this.props;
+
+        let content = null;
+
+        if (isValidating) {
+            content = this._renderSpinner();
+        } else if (isTokenValid) {
+            content = this._renderSignOut();
+        }
+
+        // else { // Sign in screen:
+        // We don't need to render any additional information.
+        // }
+
+        return (
+            <Container
+                className = 'recording-dialog'
+                style = { styles.container }>
+                <Container
+                    className = 'recording-header'
+                    style = { styles.header }>
+                    <Text
+                        className = 'recording-title'
+                        style = { styles.title }>
+                        { t('recording.authDropboxText') }
+                    </Text>
+                    <Switch
+                        disabled = { isValidating }
+                        onValueChange = { this._onSwitchChange }
+                        style = { styles.switch }
+                        value = { isTokenValid } />
+                </Container>
+                <Container
+                    className = 'authorization-panel'>
+                    { content }
+                </Container>
+            </Container>
+        );
+    }
+
+    _onSwitchChange: boolean => void;
+
+    /**
+     * Handler for onValueChange events from the Switch component.
+     *
+     * @returns {void}
+     */
+    _onSwitchChange() {
+        if (this.props.isTokenValid) {
+            this._signOut();
+        } else {
+            this._signIn();
+        }
+    }
+
+    /**
+     * Renders a spinner component.
+     *
+     * @returns {React$Component}
+     */
+    _renderSpinner() {
+        return (
+            <LoadingIndicator
+                isCompleting = { false }
+                size = 'medium' />
+        );
+    }
+
+    /**
+     * Renders the screen with the account information of a logged in user.
+     *
+     * @returns {React$Component}
+     */
+    _renderSignOut() {
+        const { spaceLeft, t, userName } = this.props;
+
+        return (
+            <Container>
+                <Container
+                    className = 'logged-in-panel'
+                    style = { styles.loggedIn }>
+                    <Container>
+                        <Text>
+                            { t('recording.loggedIn', { userName }) }
+                        </Text>
+                    </Container>
+                    <Container>
+                        <Text>
+                            {
+                                t('recording.availableSpace', {
+                                    spaceLeft,
+
+                                    // assuming 1min -> 10MB recording:
+                                    duration: Math.floor((spaceLeft || 0) / 10)
```

Comment:
Hmm this estimate seems important. How about extracting this logic into a function and explaining what's the reasoning behind it in the docs ?

Reply:
OK.